app/core/src/main/java/com/fsck/k9/notification/SendFailedNotificationController.kt

```diff
@@ -16,9 +16,16 @@ internal class SendFailedNotificationController(
         val text = ExceptionHelper.getRootCauseMessage(exception)
 
         val notificationId = NotificationIds.getSendFailedNotificationId(account)
-        val folderListPendingIntent = actionBuilder.createViewFolderListPendingIntent(
-            account, notificationId
-        )
+        val pendingIntent =
+            if (account.outboxFolderId != null) {
+                actionBuilder.createViewFolderPendingIntent(
+                    account, account.outboxFolderId!!, notificationId
+                )
+            } else {
+                actionBuilder.createViewFolderListPendingIntent(
+                    account, notificationId
+                )
+            }
```

Comment:
Kotlin supports [smart casts](https://kotlinlang.org/docs/typecasts.html#smart-casts) for read-only references. We can leverage that plus the fact that nullability is part of the type system to avoid the `!!` operator.

```kotlin
        val pendingIntent = account.outboxFolderId.let { outboxFolderId ->
            if (outboxFolderId != null) {
                actionBuilder.createViewFolderPendingIntent(account, outboxFolderId, notificationId)
            } else {
                actionBuilder.createViewFolderListPendingIntent(account, notificationId)
            }
        }
```

Reply:
You could get rid of the `if` by writing it like this:

```kotlin
        val pendingIntent = account.outboxFolderId?.let { outboxFolderId ->
            actionBuilder.createViewFolderPendingIntent(account, outboxFolderId, notificationId)
        } ?: actionBuilder.createViewFolderListPendingIntent(account, notificationId)
```

But I find this piece of code much harder to read.